base_report_to_printer/models/ir_actions_report.py

```diff
@@ -118,6 +120,29 @@ def behaviour(self):
                 result["printer_exception"] = True
         return result
 
+    def print_document_client_action(self, record_ids, data=None):
+        behaviour = self.behaviour()
+        printer = behaviour.pop("printer", None)
+        if printer.multi_thread:
+
+            @self.env.cr.postcommit.add
+            def _launch_print_thread():
+                threaded_calculation = threading.Thread(
+                    target=self.print_document_threaded,
+                    args=(self.id, record_ids, data),
+                )
+                threaded_calculation.start()
+
+            return True
+        else:
+            return self.print_document(record_ids, data=data)
+
+    def print_document_threaded(self, report_id, record_ids, data):
+        with registry(self._cr.dbname).cursor() as cr:
+            self = self.with_env(self.env(cr=cr))
```

Comment:
you should not reuse the env into another thread.....

Reply:
Oks.. 

So, I can create a new one...